dali/pipeline/executor/executor.cc

```diff
@@ -356,10 +356,20 @@ int Executor<WorkspacePolicy, QueuePolicy>::InferBatchSize(
     DALI_ENFORCE(bsps[0]->NextBatchSize() == bsps[i]->NextBatchSize(),
                  "Batch size must be uniform across an iteration");
   }
-  auto batch_size = bsps[0]->NextBatchSize();
-  assert(batch_size > 0);
-  for (auto &bsp : bsps) {
-    bsp->Advance();
+  try {
+    auto batch_size = bsps[0]->NextBatchSize();
+    assert(batch_size > 0);
+    for (auto &bsp : bsps) {
+      bsp->Advance();
+    }
+  } catch (const std::runtime_error &e) {
+    DALI_FAIL(
+        make_string("Failed to acquire the next batch size. Make sure, that DALI Pipeline is fed "
+                    "with sufficient amount of data. ", e.what()));
+  } catch (const std::out_of_range &e) {
+    DALI_FAIL(
+        make_string("No more batches in cache left. Make sure, that DALI Pipeline is fed with "
+                    "sufficient amount of data. ", e.what()));
```

Comment:
What is the cache? I think both errors should go to the same bag - the user hasn't provided enough data.